src/tests/test_kast.py

```
@@ -1,15 +1,43 @@
 from typing import Final, List, Tuple
 from unittest import TestCase
 
-from pyk.kast import KApply, KDefinition, KFlatModule, KInner, KLabel, KSequence, KSort, build_assoc
+from pyk.kast import KApply, KAtt, KDefinition, KFlatModule, KInner, KLabel, KSequence, KSort, KVariable, build_assoc
 from pyk.prelude.kbool import BOOL
 from pyk.prelude.kint import INT
 from pyk.prelude.string import STRING
 from pyk.prelude.utils import token
+from pyk.utils import FrozenDict
 
 from .utils import f, x, y, z
 
 
+class KVariableTest(TestCase):
+    TEST_DATA: Final[Tuple[Tuple[str, KInner, KInner], ...]] = (
+        ('no-sort', KVariable('Foo'), KVariable('Foo')),
+        ('sort', KVariable('Foo', sort=INT), KVariable('Foo', sort=INT)),
+        (
+            'sort-via-att',
+            KVariable('Foo', att=KAtt({'org.kframework.kore.Sort': FrozenDict(INT.to_dict())})),
+            KVariable('Foo', sort=INT),
+        ),
+        (
+            'sort-overrides-att',
+            KVariable('Foo', sort=INT, att=KAtt({'org.kframework.kore.Sort': FrozenDict(BOOL.to_dict())})),
```

In such a case, consider raising `ValueError` instead.

---

@tothtamas28 I feel that this works better with the `let` interface? Or do you feel its better to throw anyway, and introduce extra logic in let to deal with this case?

---

Yes, let's raise an exception to prevent accidental redefinition of the sort. (Similarly in `let`).

---

Done.